src/Altinn.App.Core/Internal/Auth/IAuthorizationClient.cs

```diff
@@ -50,4 +51,12 @@ Task<bool> AuthorizeAction(
     /// <param name="actions"></param>
     /// <returns></returns>
     Task<Dictionary<string, bool>> AuthorizeActions(Instance instance, ClaimsPrincipal user, List<string> actions);
+
+    /// <summary>
+    /// Retrieves roles for a user on a specified party.
+    /// </summary>
+    /// <param name="userId">The user id.</param>
+    /// <param name="userPartyId">The user party id.</param>
+    /// <returns>A list of roles for the user on the specified party.</returns>
+    Task<IEnumerable<Role>> GetUserRolesAsync(int userId, int userPartyId);
```

Comment:
I don't think we use `Async` suffix in the majority of cases, so maybe we should remove that (same goes for the telemetry trace)